x-pack/plugins/task_manager/server/task_claimers/strategy_mget.ts

```diff
@@ -357,7 +357,10 @@ async function searchAvailableTasks({
       // Task must be enabled
       EnabledTask,
       // a task type that's not excluded (may be removed or not)
-      OneOfTaskTypes('task.taskType', claimPartitions.unlimitedTypes),
+      OneOfTaskTypes(
+        'task.taskType',
+        claimPartitions.unlimitedTypes.concat(Array.from(removedTypes))
```

Comment:
Is this going to cause a starvation problem?  I don't think so, as it looks like we are removing them (marking them to `unrecognized`) at the ed of the claim, but not clear to me why this wasn't in here before.  Perhaps there was some confusion since we also have an issue to handle these via a separate stand-alone task?

This code does seem right, so I'm guessing some test was expecting these to be marked as unrecognized, and perhaps that wasn't happening before this PR ...

Reply:
>I'm guessing some test was expecting these to be marked as unrecognized, and perhaps that wasn't happening before this PR ...

Yeah there was a test in the update_by_query suite that ensured removed task types get marked as unrecognized: 
https://github.com/elastic/kibana/blob/fb33997adda561175604e68d99129fe3c64de26b/x-pack/test/plugin_api_integration/test_suites/task_manager/task_management_removed_types.ts#L79

I was able to make the test pass by adding this bit to the mget filter but I agree we should move this to its own task next via https://github.com/elastic/kibana/issues/192686